docs/generalized_distribution.md

```diff
@@ -0,0 +1,83 @@
+# Generalized distribution
+
+As of Zeebe version 8.3 we have introduced a generic way to distribute commands to other partitions.
+We felt the need to generalize this as we are seeing more feature requests that will require
+distribution.
+
+## Why do we need to distribute commands?
+
+Zeebe is a distributed system. We have multiple partitions working together to achieve high
+availability and performance.
+
+When we deploy a new process this process must be available on all
+partitions. If this is not the case we would only be able to start instances of this process on a
+single partition. By distributing it instances can be started across all partitions.
+
+## How do we distribute commands?
+
+On a high-level we use commands and event to distribute commands. The image below provides a nice
+overview of the flow of commands and events that are written during distribution.
+
+![Generalized distribution overview](assets/generalized_distribution.png)
```

Comment:
💭 I thought a bit about us using Tasks to show when we append events and commands. The way it currently reads is like Tasks that are actually Events. But for us there is some work to do on each task: Append the record. 

🔧 Let's use that in the name of those tasks: `Append Command Distribution Started`, etc.

💭 This also clarifies the word `Process` in `Process Distributed Command`. The task is to actually process that command, not something about a Process.

💭 It is a Camunda Modeling Best Practise to start Tasks with verbs instead of nouns

Reply:
> Append Command Distribution Started

I don't like this naming. This reads as if it's just appended and nothing is processed. I quite like having it the way it is to be honest. I can renamed `Process Distributed Command` to `Process the Distributed Command` to clarify it more. This way it isn't the same format as the actual commands.

> It is a Camunda Modeling Best Practise to start Tasks with verbs instead of nouns

I don't see this as a BPMN process 😄 I just use the component because it's easy to make a diagram.